code/modules/tgui_input/say_modal/typing.dm

```diff
@@ -76,30 +71,30 @@ GLOBAL_DATUM_INIT(typing_indicator, /mutable_appearance, mutable_appearance('ico
 
 /// Overrides for overlay creation
 /mob/living/create_thinking_indicator()
-	if(thinking_indicator || typing_indicator || !thinking_IC || stat != CONSCIOUS )
+	if(active_thinking_indicator || active_typing_indicator || !thinking_IC || stat != CONSCIOUS )
 		return FALSE
-	add_overlay(GLOB.thinking_indicator)
-	thinking_indicator = TRUE
+	active_thinking_indicator = mutable_appearance('icons/mob/talk.dmi', "[bubble_icon]3", TYPING_LAYER)
+	add_overlay(active_thinking_indicator)
 
 /mob/living/remove_thinking_indicator()
-	if(!thinking_indicator)
+	if(!active_thinking_indicator)
 		return FALSE
-	cut_overlay(GLOB.thinking_indicator)
-	thinking_indicator = FALSE
+	cut_overlay(active_thinking_indicator)
+	active_thinking_indicator = FALSE
 
 /mob/living/create_typing_indicator(override = FALSE)
-	if(typing_indicator || ((thinking_indicator || !thinking_IC) && !override) || stat != CONSCIOUS)
+	if(active_typing_indicator || ((active_thinking_indicator || !thinking_IC) && !override) || stat != CONSCIOUS)
 		return FALSE
-	add_overlay(GLOB.typing_indicator)
-	add_overlay(GLOB.blind_typing_indicator)
-	typing_indicator = TRUE
+	active_blind_typing_indicator = mutable_appearance('icons/mob/talk.dmi', "[bubble_icon]0", (-TYPING_LAYER), BLIND_FEATURE_PLANE, appearance_flags = KEEP_TOGETHER)
```

Comment:
This doesn't do it? @itsmeow

Reply:
Did you.. test it?

Reply:
I only tested it without blind people oops, aaand you were right, it breaks it, I'll fix it asap